### PR TITLE
Write manifest after each update.

### PR DESCRIPTION
**Pull request by ChrisHines**
Fix for #228.

### CODE REVIEW

File: cmd/gb-vendor/update.go [hunk 1]
```diff
@@ -119,12 +119,16 @@ Flags:
 				return err
 			}
```
Comment (davecheney) [inline cmd/gb-vendor/update.go]: github won't let me comment on line 73, but please move line 73 down to around 117, the sequence should be

1. fetch dependency into tempdir
2. remove old copy
3. copypath the new copy
4. write back manifest
5. wc.Destroy

**Comment by davecheney** [comment]
Thanks for this. Please review the comments i've made and I can merge this for you.

File: cmd/gb-vendor/update.go (hunk 1 above)
Comment (ChrisHines) [inline cmd/gb-vendor/update.go]: Ah yes, good catch. One moment.

File: cmd/gb-vendor/update.go [hunk 2]
```diff
@@ -100,6 +90,11 @@ Flags:
...
 
+			err = m.RemoveDependency(d)
```
Comment (davecheney) [inline cmd/gb-vendor/update.go]: I am very sorry, I told you the wrong thing, m.RemoveDependency(d) checks that the dependency is present in the manifest. It should be above the call to DeduceRepo.

File: cmd/gb-vendor/update.go [hunk 3]
```diff
@@ -112,6 +107,11 @@ Flags:
 				return err
 			}
 
+			if err := os.RemoveAll(filepath.Join(ctx.Projectdir(), "vendor", "src", filepath.FromSlash(d.Importpath))); err != nil {
```
Comment (davecheney) [inline cmd/gb-vendor/update.go]: this is fine, please don't move this

**Comment by davecheney** [comment]
Sorry, one final tweek. I'm sorry I told you the wrong thing earlier.

**Comment by ChrisHines** [comment]
No worries.

**Pull request merged**